src/markets/kraken.rs

```diff
@@ -0,0 +1,150 @@
+use crate::markets;
+use crate::markets::TradingPair;
+use chrono::{DateTime, NaiveDateTime, Utc};
+use serde::de::Error;
+use serde::Deserialize;
+use std::convert::TryFrom;
+
+/// Fetch OHLC (open-high-low-close) data
+/// More info here: https://www.kraken.com/features/api
+pub async fn get_ohlc(trading_pair: TradingPair) -> anyhow::Result<markets::Ohlc> {
+    let trading_pair_code = get_trading_pair_code(trading_pair);
+
+    // Interval used when fetching the ohlc data from Kraken.
+    // The data returned will contain segments according to the interval.
+    // The highest frequency time interval for OHLC data is 1 minute, possible values:
+    // 1 (default), 5, 15, 30, 60, 240, 1440, 10080, 21600
+    let time_interval = 30;
+    // By passing in a timestamp far in the futrue we reduce the API to return only the last OHLC value
+    let since = 2_147_483_647;
+
+    let request_url = format!("https://api.kraken.com/0/public/OHLC?pair={trading_pair}&interval={time_interval}&since={since}",
+                              trading_pair = trading_pair_code,
+                              time_interval = time_interval,
+                              since = since,
+    );
+
+    let response = reqwest::get(&request_url)
+        .await?
+        .json::<OhlcResponse>()
+        .await?;
+
+    let ohlc = response.result.xbtdai;
```

Comment:
Yeah... this is partly because I already introduced abstraction that probably should not be there yet. I would keep it as is at the moment (some things were already removed again), and care about it again once we extend it.